runtime/runtime-params-estimator/emu-cost/counter_plugin/counter.c

```diff
@@ -83,15 +101,39 @@ static void vcpu_syscall(qemu_plugin_id_t id, unsigned int vcpu_index,
     }
 }
 
+static void vcpu_ret_syscall(qemu_plugin_id_t id, unsigned int vcpu_index,
+                             int64_t num, int64_t ret)
+{
+    if (counting && ret > 0) {
+        switch (num) {
+            case 0: // sys_read
+            case 17: // sys_pread64
+            case 19: // sys_readv
+            case 295: // sys_preadv
+                __atomic_fetch_add(&read_count, ret, __ATOMIC_SEQ_CST);
+                break;
+            case 1: // sys_write
+            case 18: // sys_pwrite64
+            case 20: // sys_writev
+            case 296: // sys_pwritev
+                __atomic_fetch_add(&write_count, ret, __ATOMIC_SEQ_CST);
```

Comment:
we record number of cpu instructions also via a sys_write, will this cause counting cpu instructions also increase write_count?

Reply:
It doesn't work this way. We start counting by sys_read, instructions are counted by QEMU plugin, we only retrieve the final count with another `sys_read`.